pkg/catalog/catalog.go

```diff
@@ -2260,7 +2264,17 @@ func (c *Catalog) runBackgroundTaskSteps(repository *graveler.RepositoryRecord,
 			if err != nil {
 				log.WithError(err).WithField("step", step.Name).Errorf("Catalog background task step failed")
 				task.Done = true
-				task.Error = err.Error()
+				apierrors.HandleAPIErrorCallback(ctx, log, nil, nil, err, func(w http.ResponseWriter, r *http.Request, code int, v interface{}) {
```

Comment:
This seems like the wrong place to translate error into user facing **api** error. 
It should be at the controller level.